File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
       int maxSoFar = nums[0]; 
        int currentMax = nums[0]; 
        for (int i = 1; i < nums.size(); ++i) {
            currentMax = max(nums[i], currentMax + nums[i]);
            maxSoFar = max(maxSoFar, currentMax);
        }

        return maxSoFar;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if(nums.length == 1) return nums[0];
         int sum = 0,ans= Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            sum += nums[i];
           if(sum > ans){
            ans = sum;
           }
            
           if(sum < 0){
            sum = 0;
           }
        }
        return ans;
    }
}